src/main/scala/org/broadinstitute/hail/driver/ExportVCF.scala

```diff
@@ -207,7 +207,28 @@ object ExportVCF extends Command {
           sb.append("GT:AD:DP:GQ:PL")
         gs.foreach { g =>
           sb += '\t'
-          sb.append(g)
+
+          sb.append(g.gt.map { gt =>
+            val p = Genotype.gtPair(gt)
+            s"${ p.j }/${ p.k }"
+          }.getOrElse("./."))
+          sb += ':'
+          sb.append(g.ad.map(_.mkString(",")).getOrElse("."))
+          sb += ':'
+          sb.append(g.dp.map(_.toString).getOrElse("."))
+          sb += ':'
+          sb.append(g.gq.map(_.toString).getOrElse("."))
+          sb += ':'
+          if (!g.isDosage)
+            sb.append(g.pl.map(_.mkString(",")).getOrElse("."))
+          else
+            sb.append(g.dosage.map(_.mkString(",")).getOrElse("."))
+
+          val regex = """(?::\.)+$""".r
+          val regexMatch = regex.findFirstMatchIn(sb)
+          regexMatch.foreach { m =>
+            sb.delete(m.start, m.end)
+          }
```

Comment:
Just define some helper functions.  Can't do that in your version since you have to increment nNulls.  Anyway, we're getting into the weeds.  Alex's got this.